components/ILIAS/AuthApache/src/UsernameProvider/UsernameProviderFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\ApacheAuth\UsernameProvider;
+
+final class UsernameProviderFactory
+{
+    /**
+     * @param list<class-string<UsernameProvider>> $class_names
+     * @return list<UsernameProvider>
+     */
+    public function fromClassNames(array $class_names): array
+    {
+        $instances = [];
+        foreach ($class_names as $class) {
+            if (!\is_string($class) || !class_exists($class)) {
```

Comment:
```suggestion
            if (!\is_string($class) || !class_exists($class, false)) {
```

As we manually require the file with the following call, should we consider disable autoloading for class_exists check?

```php
$resolver = new UsernameResolver($factory->fromClassNames(
    require CollectUsernameProvidersObjective::PATH()
));
```

Reply:
We don't require a class here. We require the artifact which returns a list of class names of provider classes, hopefully implementing the correct interface.

If we disable the autoloading here and a provider has not been auto-loaded yet, this might result in a PHP error.